test/suite/test_config11.py

```diff
@@ -0,0 +1,110 @@
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_config11.py
+#   Test the debug configuration setting on the session which evicts
+#   pages as they are released.
+#
+
+import wttest
+from wiredtiger import stat
+from wtscenario import make_scenarios
+from wtdataset import SimpleDataSet
+
+class test_config11(wttest.WiredTigerTestCase):
+    conn_config = 'statistics=(fast)'
+    session_config = 'isolation=snapshot'
+    key_format_values = [
+        ('column', dict(key_format='r')),
+        ('integer_row', dict(key_format='i')),
+    ]
+
+    scenarios = make_scenarios(key_format_values)
+
+    def test_config11(self):
+        uri = 'table:test_config11'
+
+        ds = SimpleDataSet(
+            self, uri, 0, key_format=self.key_format, value_format="S", config='log=(enabled=false)')
+        ds.populate()
+
+        # Pin oldest and stable timestamps to 10.
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(10) +
+            ',stable_timestamp=' + self.timestamp_str(10))
+
+        # Retrieve the maximum cache size.
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        max_cache_size = stat_cursor[stat.conn.cache_bytes_max][2]
+        stat_cursor.reset()
+
+        # Insert some values to fill up the cache, but not exceeding 80% of it (we set a margin of
+        # 10% to be sure).
+        value = 'abcd' * 1000
+        current_cache_usage_perc = 0
+        threshold_cache_perc = 70
+        i = 1
+
+        s = self.conn.open_session()
+        cursor = s.open_cursor(uri)
+        while(current_cache_usage_perc < threshold_cache_perc):
+            cursor[i] = value + str(i)
+            i = i + 1
+
+            # Retrieve the current cache usage.
+            current_cache_usage_perc = stat_cursor[stat.conn.cache_bytes_inuse][2] * 100 / max_cache_size
+            stat_cursor.reset()
+
+        cursor.close()
+
+        # Checkpoint should not trigger eviction, it still might reduce the cache usage by a small
+        # amount.
+        s.checkpoint()
```

Comment:
@sulabhM @mm-ng , I saw in the first version of the test, we perform a checkpoint. Was there any specific reason for doing a checkpoint?
Since we don't expect eviction to be triggered, I added a test just below. We might as well remove it if performing checkpoint is not required for this test.

Reply:
To make the content is the cache clean. If you don't checkpoint, the cache is all dirty and eviction will trigger at 5% dirty by itself. We wanted to fill the cache with clean content, such that we can use the newly added flag to trigger eviction.

Reply:
So when we are filling the cache close to 80%, it's already too late isn't it? The threshold of 5% is exceeded very quickly and eviction for dirty content is already being executed.
Should I fill the cache up to 5% to avoid this at the beginning of the test?

Reply:
I have explicitly configured a higher threshold for dirty content [here](https://github.com/wiredtiger/wiredtiger/pull/7515/commits/2052a9bf80544fe37874b38a576f054292930841#diff-f1e790183032bc0e6d3042a62d62525a2867f3f486849e65da212b1e5e0443d6R40).